openprescribing/data/rxdb/labelled_matrix.py

```diff
@@ -72,28 +77,26 @@ def create_row_grouper(input_labels, label_map):
     """
     input_label_index = {label: i for i, label in enumerate(input_labels)}
 
-    # Build a mapping from output labels to the list of input row indexes which are
-    # mapped to that label
-    output_labels = defaultdict(list)
-    for input_label, output_label in label_map:
-        input_indexes = output_labels[output_label]
-        try:
-            input_index = input_label_index[input_label]
-        except KeyError:
-            # It's convenient for our purposes if missing input row labels don't throw
-            # errors but are treated as, effectively, zero-valued rows. It means we ask
-            # e.g. what the total prescribing is for practices X, Y and Z and get the
-            # right answer even if practice Z hasn't prescribed at all and isn't present
-            # in the input.
-            continue
-        input_indexes.append(input_index)
+    output_labels = []
+    output_label_groups = []
+    for output_label, input_label_group in label_map:
+        output_labels.append(output_label)
+        output_label_groups.append(
+            [
+                input_label_index[label]
+                for label in input_label_group
+                # We filter any non-matching labels (see `group_rows()` docstring for
```

Comment:
Minor nit: as we're in the body of `group_rows()` we don't need the function name here -- thinking about the name of of the function I was reading broke my flow slightly.